include/RAJA/pattern/forall.hpp

```diff
@@ -259,7 +246,20 @@ RAJA_INLINE void forall(ExecPolicy<SegmentIterPolicy, SegmentExecPolicy>,
 /*!
  ******************************************************************************
  *
- * \brief Generic dispatch over  with icount
+ * \brief The RAJA::policy_by_value_interface forall functions provide an interface with
+ *        value-based policies. It also enforces the interface and performs
+ *        static checks as well as triggering plugins and loop body updates.
+ *
+ ******************************************************************************
+ */
+inline namespace policy_by_value_interface
```

Comment:
I made this an inline namespace to make what happens in the generic forall more clear, does that seem reasonable?

Reply:
I think that makes sense.

Reply:
I like this, when working on adding the resource argument to forall, I found that it wasn't very clear initially what differentiated these calls from the others.

Reply:
It feels a bit odd to use an inline namespace this way, but it certainly clarifies the way the file flows.  The thing that makes it feel odd to me is that this now means this namespace is part of our public interface.  On balance I think I'm good with it, but might prefer to make it clear things are never meant to be called from outside by referencing this namespace.